app/code/Magento/Inventory/Controller/Adminhtml/Stock/StockSourceLinkProcessor.php

```diff
@@ -37,34 +38,34 @@ class StockSourceLinkProcessor
     private $getAssignedSourcesForStock;
 
     /**
-     * @var AssignSourcesToStockInterface
+     * @var StockSourceLinksSaveInterface
      */
-    private $assignSourcesToStock;
+    private $stockSourceLinksSave;
 
     /**
-     * @var UnassignSourceFromStockInterface
+     * @var StockSourceLinksDeleteInterface
      */
-    private $unassignSourceFromStock;
+    private $stockSourceLinksDelete;
 
     /**
      * @param SearchCriteriaBuilder $searchCriteriaBuilder
      * @param StockSourceLinkFactory $stockSourceLinkFactory
      * @param GetAssignedSourcesForStockInterface $getAssignedSourcesForStock
-     * @param AssignSourcesToStockInterface $assignSourcesToStock
-     * @param UnassignSourceFromStockInterface $unassignSourceFromStock
+     * @param StockSourceLinksSaveInterface $stockSourceLinksSave
+     * @param StockSourceLinksDeleteInterface $stockSourceLinksDelete
      */
     public function __construct(
         SearchCriteriaBuilder $searchCriteriaBuilder,
         StockSourceLinkFactory $stockSourceLinkFactory,
         GetAssignedSourcesForStockInterface $getAssignedSourcesForStock,
```

Comment:
don't use `GetAssignedSourcesForStockInterface ` in this class.

Because from one side you use it for easily retrieve  Sources assigned: 

`$assignedSources = $this->getAssignedSourcesForStock->execute($stockId);`
But after that you added own method to build SourceLinks based on that:

` private function getStockSourceLinks(array $sourceCodeList, $stockId): array`

just remove this dependency and use GetSourceLinksInterface instead:
```
interface GetSourceLinksInterface
{
    /**
     * Find StockSourceLink list by given SearchCriteria
     *
     * @param \Magento\Framework\Api\SearchCriteriaInterface $searchCriteria
     * @return \Magento\InventoryApi\Api\Data\StockSourceLinkSearchResultsInterface
     */
    public function execute(SearchCriteriaInterface $searchCriteria): StockSourceLinkSearchResultsInterface;
}
```
which returns `StockSourceLinkInterface`. Thus, no need to build it by yourself.